### PR TITLE
Implement code actions for "Add to dictionary", "Disable rule" and "Hide false positive"

### PR DESCRIPTION
**Pull request by abougouffa**
# Summary

This PR adds implementation of the LSP actions:
- [`_ltex.addToDictionary`](https://valentjn.github.io/ltex/ltex-ls/server-usage.html#_ltexaddtodictionary-client)
- [`_ltex.disableRules`](https://valentjn.github.io/ltex/ltex-ls/server-usage.html#_ltexdisablerules-client)
- [`_ltex.hideFalsePositives`](https://valentjn.github.io/ltex/ltex-ls/server-usage.html#_ltexhidefalsepositives-client)

The actions can be executed by calling `lsp-execute-code-action` or by clicking on the `lsp-ui` elements:

Screenshot_20220805_181957

# Idea

The idea is to define an internal plist for the dictionary (same for disabled rules and false positives) `lsp-ltex--stored-dictionary` in which, words will be added when the user apply the LSP action "Add 'word' to dictionary", this plist gets automatically stored under `lsp-ltex-user-rules-path`, and it is loaded at startup.

To allow backward compatibility, the user configured `lsp-ltex-dictionary` variable is read and combined with the created `lsp-ltex--stored-dictionary`, to provide the internal variable `lsp-ltex--combined-dictionary` which is the actual data sent to LTeX-LS via `ltex.dictionary`.

# Related issues

Fixes: https://github.com/emacs-languagetool/lsp-ltex/issues/4 https://github.com/emacs-languagetool/lsp-ltex/issues/16

# Refs

- https://valentjn.github.io/ltex/ltex-ls/server-usage.html

### CODE REVIEW

File: lsp-ltex.el
Comment (jcs090218) [inline lsp-ltex.el:373]: Maybe revert the order? So it's matches to their [settings](https://valentjn.github.io/ltex/settings.html) page.

**Review by jcs090218 — APPROVED** [review]
This is awesome! Thanks! ❤️ 👍 One nit (see the review).

I noticed right after you do one of the code action, it does not apply to the current session. You would have to restart workspace in order to have these values applied. Any idea to improve this behaviour? 😕

File: lsp-ltex.el
Comment (abougouffa) [reply lsp-ltex.el:373]: Ok!

**Comment by abougouffa** [comment]
@jcs090218 Thank for the review,

After applying an action, let's say "adding a word to dictionary", it won't take effect immediately, it needs to wait for `ltex-ls` to recheck the text (taking into consideration the new added rule). But AFAK, it doesn't demand restarting the session.

The behavior depends also on `lsp-ltex-check-frequency`, it takes more or less long when it is set to `"save"` for example.

**Comment by jcs090218** [comment]
> After applying an action, let's say "adding a word to dictionary", it won't take effect immediately, it needs to wait for ltex-ls to recheck the text (taking into consideration the new added rule). But AFAK, it doesn't demand restarting the session.

Ah!! Thanks for the letting me know! I didn't realize that. 👍 

One more concern, I know this will generate the following three files

* `lsp-ltex--stored-dictionary.el`
* `lsp-ltex--stored-disabled-rules.el`
* `lsp-ltex--stored-hidden-false-positives.el`

Can we shorten the name of these three files? 😕 I think it would be easier for us to see what each file does. How about without the prefix `lsp-ltex--` and the extensions `.el`?

* `stored-dictionary`
* `stored-disabled-rules`
* `stored-hidden-false-positives`

WDYT? Thanks! ❤️

**Comment by abougouffa** [comment]
Done!

Please do not merge right now, I will push some additional commits, I'm testing them locally.

**Comment by abougouffa** [comment]
Done, I've pushed some additional stuff:

- Remove obsolete option `"ltex.java.forceTrySystemWide"` (if set, `ltex.java.path` will do the work)
- Rename option `ltex.ltex-ls.languageToolHttpServerUri` to [`ltex.languageToolHttpServerUri`](https://valentjn.github.io/ltex/settings.html#ltexlanguagetoolhttpserveruri)
- Add `t` to indicate that the option is a boolean
- Add some missing options
  - [ltex.languageToolOrg.username](https://valentjn.github.io/ltex/settings.html#ltexlanguagetoolorgusername)
  - [ltex.languageToolOrg.apiKey](https://valentjn.github.io/ltex/settings.html#ltexlanguagetoolorgapikey)
  - [ltex.ltex-ls.path](https://valentjn.github.io/ltex/settings.html#ltexltex-lspath)
  - [ltex.completionEnabled](https://valentjn.github.io/ltex/settings.html#ltexcompletionenabled) (needs integration with something like `company`, TBD).
  - [ltex.statusBarItem](https://valentjn.github.io/ltex/settings.html#ltexstatusbaritem) (needs implementation)

**Comment by abougouffa** [comment]
Waiting for your review @jcs090218 !

Thank you for all the work you are doing!

**Comment by jcs090218** [comment]
Wow, that's a lot of improvements! ❤️ 🎉 

> Done, I've pushed some additional stuff:

You have also updated the documentation! That's so nice! I appreciated! 💯 

> Thank you for all the work you are doing!

No, I should say thank you! ❤️ Sorry I don't have much time to improve this package by seeing my last commit was pretty long ago. 😓 

---

Merged! Thank you! 💯

**Comment by djgoku** [comment]
I know this is `lsp-ltex` but can any of this be adapted for `eglot-ltex` too?

**Comment by abougouffa** [comment]
Hello @djgoku,

I'm using Eglot from a while, but for the moment, I didn't take a look to how integrate these features with Eglot. I've taken a quick look at Eglot's source code, but I didn't find a trivial way to do it.

**Pull request merged**
